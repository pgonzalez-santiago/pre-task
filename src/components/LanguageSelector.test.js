import React from 'react'
import ConnectedLanguageSelector, { LanguageSelector } from './LanguageSelector'
import { mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { INITIAL_STORE } from '../../test/mocks.js'

const changeLangMock = jest.fn()

/* Material-UI */
import Switch from '@material-ui/core/Switch'

describe('Language Selector component', () => {

  it('renders correctly with error', () => {
    const wrapper = shallow(
      <LanguageSelector
        changeLang={changeLangMock}
        currentLang='en'
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('changeLang is called with es', () => {
    const wrapper = shallow(
      <LanguageSelector
        changeLang={changeLangMock}
        currentLang='en'
      />
    )

    wrapper.find(Switch).at(0).props().onChange()

    expect(changeLangMock).toBeCalledWith('es')
  })

  it('changeLang is called with en', () => {
    const wrapper = shallow(
      <LanguageSelector
        changeLang={changeLangMock}
        currentLang='es'
      />
    )

    wrapper.find(Switch).at(0).props().onChange()

    expect(changeLangMock).toBeCalledWith('en')
  })
})

describe('ConnectedLanguageSelector', () => {
  const mockStore = configureStore()
  let wrapper
  let store

  store = mockStore(INITIAL_STORE)

  wrapper = mount(
      <Provider store={store}>
        <ConnectedLanguageSelector />
      </Provider>
  )

  it('Stays the same as before with id and application', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Actions called properly', () => {
    const ChildComponent = wrapper.find(LanguageSelector)
    expect(ChildComponent.props().changeLang('en')).toEqual({ "payload": "en", "type": "lang/set" })
  })
})
