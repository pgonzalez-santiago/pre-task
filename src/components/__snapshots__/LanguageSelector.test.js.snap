// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedLanguageSelector Stays the same as before with id and application 1`] = `ReactWrapper {}`;

exports[`Language Selector component renders correctly with error 1`] = `ShallowWrapper {}`;
